dev/io.openliberty.microprofile.openapi.ui.internal_fat/fat/src/io/openliberty/microprofile/openapi/ui/internal/fat/tests/UICustomPathTest.java

```diff
@@ -0,0 +1,159 @@
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.*;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.BrowserWebDriverContainer;
+import org.testcontainers.containers.BrowserWebDriverContainer.VncRecordingMode;
+import org.testcontainers.containers.VncRecordingContainer.VncRecordingFormat;
+
+import java.io.FileWriter;
+import java.time.Duration;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * A Test of UI Path rewriting
+ */
+@RunWith(FATRunner.class)
+public class UICustomPathTest {
+
+    /** Wait for "long" tasks like initial page load or making a test request to the server */
+    private static final Duration LONG_WAIT = Duration.ofSeconds(30);
+
+    /** Wait for "short" tasks which shouldn't need to wait for anything, like selecting a drop down */
+    private static final Duration SHORT_WAIT = Duration.ofSeconds(5);
+
+    public static final String APP_NAME = "app";
+
+    public static final String PROPERTY_NAME= "mp.openapi.extensions.liberty.ui.path";
+
+    public static final String PATH_VALUE= "/foo/bar";
+
+    @Server("openapi-ui-test")
+    public static LibertyServer server;
+
+    @Rule
+    public BrowserWebDriverContainer<?> chrome = new BrowserWebDriverContainer<>().withCapabilities(new ChromeOptions())
+            .withAccessToHost(true)
+            .withRecordingMode(VncRecordingMode.RECORD_FAILING,
+                    Props.getInstance().getFileProperty(Props.DIR_LOG),
+                    VncRecordingFormat.MP4)
+            .withLogConsumer(new SimpleLogConsumer(UICustomPathTest.class, "selenium-driver"));
+
+    private RemoteWebDriver driver;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                .addClass(TestResource.class);
+
+        ShrinkHelper.exportDropinAppToServer(server, war, DeployOptions.SERVER_ONLY);
+
+        ServerFileUtils serverFileUtils = new ServerFileUtils();
+
+        String bootProps = serverFileUtils.getServerFileLoc(server) + "/bootstrap.properties";
+        // append to bootstrap.properties
+        FileWriter writer = new FileWriter(bootProps, true);
+        writer.append(PROPERTY_NAME + "=" + PATH_VALUE);
+        writer.append(System.getProperty("line.separator"));
+        writer.close();
+
+        String jvmOptions = serverFileUtils.getServerFileLoc(server)+"/jvm.options";
+        writer = new FileWriter(jvmOptions, true);
+        writer.append("-Dcom.ibm.ws.beta.edition" + "=" + "true");
+        writer.close();
```

Comment:
Usually we would just write the bootstrap.properties file ourselves and put it in `/publish/servers/<servername>` if the content doesn't need to be dynamic.

Here it might make sense for it to be dynamic if we write more tests, so we could do it this way, but in that case we'll also have to be a little bit careful to avoid the config from one test contaminating another test.

If we do want to write a properties file directly, I would generally use `Properties.load()` and `.store()` and you should always use the try-with-resources syntax to close writers where possible.